README.md

```diff
@@ -11,130 +12,128 @@ The framework will help you to:
 
 ## Getting Started
 
-The following overall structure for your charm directory is recommended:
+Charms written using the operator framework are just Python code. The intention
+is for it to feel very natural for somebody used to coding in Python, and
+reasonably easy to pick up for somebody who might be a domain expert but not
+necessarily a pythonista themselves.
 
-```
-.
-├── config.yaml
-├── metadata.yaml
-├── mod/
-├── lib/
-│   └── ops -> ../mod/operator/ops
-├── src/
-│   └── charm.py
-└── hooks/
-    ├── install -> ../src/charm.py
-    └── start -> ../src/charm.py  # for k8s charms per below
-```
-
-The `mod/` directory should contain the operator framework dependency as a git
-submodule:
+The dependencies of the operator framework are kept as minimal as possible;
+currently that's Python 3.5 or greater, and `PyYAML` (both are included by
+default in Ubuntu's cloud images from 16.04 on).
 
-```
-git submodule add https://github.com/canonical/operator mod/operator
-```
+If you're new to the world of juju and charms, you should probably dive into our
+[tutorial](/TBD).
 
-Then symlink from the git submodule for the operator framework into the `lib/`
-directory of your charm so it can be imported at run time:
+If you know about juju, and have written charms that didn't use the operator
+framework (be it with reactive or without), we have an [introduction to the
+operator framework](/TBD) just for you.
 
-```
-ln -s ../mod/operator/ops lib/ops
-```
+If you've gone through the above already and just want a refresher, or are
+really impatient and need to dive in, feel free to carry on down.
 
-Other dependencies included as git submodules can be added in the `mod/`
-directory and symlinked into `lib/` as well.
+## A Quick Introduction
 
-You can sync subsequent changes from the framework and other submodule
-dependencies by running:
+Operator framework charms are just Python code. The entry point to your charm is
+a particular Python file. It could be anything that makes sense to your project,
+but let's assume this is `src/charm.py`. This file must be executable (and it
+must have the appropriate shebang line).
 
-```
-git submodule update
-```
+This file must be symlinked from `dispatch` (as of juju 3.8; otherwise from
+`hooks/install`, and also from `hooks/start` if it's for k8s).
```

Comment:
We should also advise hooks/upgrade-charm, since that is the other way that you 'get new code' onto a system.

And its Juju 2.8.

Proposal:
```
This file must be symlinked from `dispatch` (supported by Juju 2.8; otherwise from `hooks/install` and `hooks/upgrade-charm`, and `hooks/start` if it is a k8s charm.)
```

Reply:
Juju 4.8, got it.